main.js

```diff
@@ -1,56 +1,302 @@
+                        userState.textContent = document.userform.state.value;
+                        document.getElementById('print');
+                        printParent.appendChild(userState);
+
+
+
+    var userZipcode = document.createElement('div');
+                        userZipcode.textContent = document.userform.zipcode.value;
+                        document.getElementById('print');
+                        printParent.appendChild(userZipcode);
+
+
+
+    var userPhone= document.createElement('div');
+                    userPhone.textContent = document.userform.phone.value;
+                    document.getElementById('print');
+                    printParent.appendChild(userPhone);
+
+    var userEmail = document.createElement('div');
+                        userEmail.textContent = document.userform.email.value;
+                        document.getElementById('print');
+                        printParent.appendChild(userEmail);
+
+
 }
 
+var userform = document.querySelector("form")
+        userform.addEventListener('submit', check);
+
+
+
+
+function validateFirstName() 
+{
+    var first = document.getElementById('firstname').value;
+   
+        if (!first.match(/^[a-zA-Z]+$/)) 
+
+    {
+        alert('Only letters are allowed for first name.');
+        return false;
+    }
+    
+    return true;
+}
+
+function validateLastName() 
+{
+    var last = document.getElementById('lastname').value;
+   
+        if (!last.match(/^[a-zA-Z]+$/)) 
+
+    {
+        alert('Only letters are allowed for last name.');
+        return false;
+    }
+    
+    return true;
+}
+
+
+
+function validateAddress() {
+
+    var address = document.userform.address.value;
+     
+    if(address !== null && address !== "")
+    {
+
+        return true;
+    }
+    else{
+        alert("Address is required");
+        return false;
+    }
+
+
+}
+
+function validateCity() {
+
+    var city = document.userform.city.value;
+     
+    if(city!== null && city !== "")
+    {
+
+        return true;
+    }
+    else{
+        alert("City is required");
+        return false;
+    }
+
+
+}
+
+function validateState() {
+
+    var state = document.userform.state.value;
+    var lettersOnly = document.userform.state.value;
+
+    if(state.length === 2 && lettersOnly.match(/^[a-zA-Z]+$/))
+
+    {
+
+        return true;
+    }
+
+        else{
+        alert("2 letters required for state");
+        return false;
+    }
+}
+
+
+
+function validateZipcode() {
+    var zipcode = document.userform.zipcode.value;
+    var numbersOnly = document.userform.zipcode.value;
+
+    if(zipcode.length === 5 && numbersOnly.match(/^\d+/)) 
+    {
+            return true; 
+}
+        else { 
+        alert("5 numeric characters required for zipcode");
+        return false;
+
+    }
+
+}
+
+function validatePhoneNumber() {
+    var phonenum= /^\(?([0-9]{3})\)?[-. ]?([0-9]{3})[-. ]?([0-9]{4})$/;
+    if(phone.value.match(phonenum)) {
```

Comment:
i'm not sure where this `phone` value comes from. it behaves like we'd expect but it's not declared anywhere and i honestly don't know how it's working, but make sure to get a `phone` value from the form like you do for everything else

Reply:
this function is now in alignment with html type and getting value by same process as all other functions